test/dynamo/test_misc.py

```diff
@@ -1382,14 +1379,14 @@ def _fn(a, b, func=func):
             get_test_fn(func=min),
             2,
             expected_ops=1,
-            expected_ops_dynamic=ifdynstaticdefault(1, 14),
+            expected_ops_dynamic=ifdynstaticdefault(1, 12),
         )
         torch._dynamo.testing.standard_test(
             self,
             get_test_fn(func=max),
             2,
             expected_ops=1,
-            expected_ops_dynamic=ifdynstaticdefault(1, 17),
```

Comment:
Test case contains:
```
lst = [a.shape[0] + 1, 8, a.shape[0]]
x = func(lst)
y = func(iter(lst))
z = func(tuple(lst))
w = func(*lst)
return a + (x + y + z + w)
```
where func is max.

here the ShapeEnv seems to say a.shape[0] has range [7, inf], so all size calls resolve to s0 + 1, and get hash consed. 8 sym_max calls and 2 (size() + getitem) calls are removed.